presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -561,17 +562,23 @@ protected Scope visitCall(Call node, Optional<Scope> scope)
             return createAndAssignScope(node, scope);
         }
 
-        private void validateColumns(Statement node, RelationType descriptor)
+        private void validateColumns(Statement node, RelationType descriptor, List<String> columnAliases)
         {
             // verify that all column names are specified and unique
             // TODO: collect errors and return them all at once
+            boolean hasColumnAliases = !columnAliases.isEmpty();
+            if (hasColumnAliases && columnAliases.size() != descriptor.getVisibleFieldCount()) {
+                throw new SemanticException(MISMATCHED_COLUMN_ALIASES, node, "Column alias list has %s entries but subquery has %s columns", columnAliases.size(), descriptor.getVisibleFields().size());
```

Comment:
I mean: once you query `descriptor.getVisibleFieldCount()` but then you use `descriptor.getVisibleFields().size()` to produce error message. Of course, they are equivalent, but why not use same method in both places?